fix: Breaking install due to missing hook & setup server side CI (LAN-720)

marination opened the pull request:
- Hook `landa_custom_fields` is empty and so `create_custom_fields` gets an empty list instead of dict, which breaks install
- CI to ensure we're not merging breaking (literally) changes
- Custom API to complete setup wizard for landa test site before creating fixtures. Only invoked if site has testing enabled or is in a CI environment

marination committed:
fix: Use fallback dict as `landa_custom_fields` is empty in hooks

- Installs break as create_custom_fields gets an empty list instead of dict

marination committed:
chore: Setup Server side test CI to check successfull install of LANDA

barredterra commented on .github/workflows/ci.yml:
```diff
@@ -0,0 +1,83 @@
+
+    services:
+      mariadb:
+        image: mariadb:10.3
```
```suggestion
        image: mariadb:10.5
```

We're running 10.5 in production

barredterra commented on .github/workflows/ci.yml:
```diff
@@ -0,0 +1,83 @@
+      - name: Setup Node
+        uses: actions/setup-node@v2
+        with:
+          node-version: 12
```
```suggestion
          node-version: 14
```

According to the docs, frappe v13 depends on node v14

marination replied:
big yikes my bad

marination committed:
fix: mariadb version

Co-authored-by: Raffael Meyer <14891507+barredterra@users.noreply.github.com>

marination committed:
fix: Node version as per docs

Co-authored-by: Raffael Meyer <14891507+barredterra@users.noreply.github.com>

marination committed:
chore: Right usage of `bench execute`

marination committed:
fix: Complete setup wizard in `after_install`

- Landa specific code can only run after install
- Complete setup wizard in `after_install`, call API
- Check if test site or for CI. only then run completion method

marination committed:
Merge branch 'version-13-hotfix' into successfull-installs

marination committed:
chore: Skip test record creation

- breaks as erpnext/frappe records are being created without landa specific mandatory data

barredterra commented:
@marination looks like there's still a redis error while completing setup wizard. Should we ignore or fix this?

marination commented:
I could not replicate this locally. Can you try `creating a test site > installing erpnext > enabling tests for site > installing landa` and see ? If that's too much I think we can ignore it. I'll take a look at it when we push tests for this app

marination committed:
fix: Start bench

- Done to fix redis conn error that appeared during setup wizard completion

marination committed:
Merge branch 'version-13-hotfix' into successfull-installs

marination committed:
chore: Pair command with bench start

barredterra committed:
Merge branch 'version-13-hotfix' into successfull-installs

barredterra commented:
:tada: This PR is included in version 0.2.0 :tada:

The release is available on [GitHub release](https://github.com/alyf-de/landa/releases/tag/v0.2.0)

Your **[semantic-release](https://github.com/semantic-release/semantic-release)** bot :package::rocket:
